ruby/ext/google/protobuf_c/message.c

```diff
@@ -910,14 +910,22 @@ static VALUE Message_freeze(VALUE _self) {
  *     Message.[](index) => value
  *
  * Accesses a field's value by field name. The provided field name should be a
- * string.
+ * string or a symbol.
  */
 static VALUE Message_index(VALUE _self, VALUE field_name) {
   Message* self = ruby_to_Message(_self);
   const upb_FieldDef* field;
 
-  Check_Type(field_name, T_STRING);
-  field = upb_MessageDef_FindFieldByName(self->msgdef, RSTRING_PTR(field_name));
+  if (TYPE(field_name) == T_STRING) {
+    name = RSTRING_PTR(field_name);
+  } else if (TYPE(field_name) == T_SYMBOL) {
+    name = RSTRING_PTR(rb_id2str(SYM2ID(field_name)));
+  } else {
+    rb_raise(rb_eArgError,
+             "Expected string or symbols as hash keys when accessing a field's "
```

Comment:
Let's not use the word "hash" here. It implies that there is a Ruby Hash object underneath, which there is not.

I think the error message words if you just remove the word "hash."